core/src/test/scala/unit/kafka/cluster/PartitionTest.scala

```diff
@@ -1545,6 +1556,160 @@ class PartitionTest {
     assertEquals(Set(), Metrics.defaultRegistry().allMetrics().asScala.keySet.filter(_.getType == "Partition"))
   }
 
+  @Test
+  def testPartitionAssignmentStatuses(): Unit = {
+    testPartitionAssignmentStatus(
```

Comment:
Could we just split this out into separate test cases and get rid of `withLocalSetup`?

Reply:
Sure :)

Reply:
I'll try to use parameterized to see if that works better.